docs/source/examples/flow_past_cylinder.rst

```diff
@@ -0,0 +1,252 @@
+
+.. code:: python
+
+    def create_particles(self):
+        fluid = self._create_fluid()
+        solid = self._create_solid()
+        outlet = self._create_outlet()
+        inlet = self._create_inlet()
+        wall = self._create_wall()
+
+        ghost_inlet = self.iom.create_ghost(inlet, inlet=True)
+        ghost_outlet = self.iom.create_ghost(outlet, inlet=False)
+
+        particles = [fluid, inlet, outlet, solid, wall]
+        if ghost_inlet:
+            particles.append(ghost_inlet)
+        if ghost_outlet:
+            particles.append(ghost_outlet)
+
+        self.scheme.setup_properties(particles)
+        self._set_wall_normal(wall)
+
+        if self.io_method == 'hybrid':
+            fluid.uag[:] = umax
+            fluid.uta[:] = umax
+            outlet.uta[:] = umax
+
+        return particles
+
+The particle arrays ``ghost_inlet`` and ``ghost_outlet`` are generated by
+the IOM depending upon the type of IOM subclass used. The properties
+:math:`uag`, :math:`uta` are the time average and velocity array in :math:`x`
+direction at t=0.
+
+Creating updater
+~~~~~~~~~~~~~~~~~
+
+The purpose of the updater is to remove particle from ``inlet`` and add them
+to ``fluid`` whenever a particle crosses the inlet-outlet interface. Similarly,
+it is done in case of the ``oulet``. It also adds new particle to ``inlet`` as required and remove a particle from the simulation when they flow past ``outlet``.
+
+.. code:: python
+
+    def create_inlet_outlet(self, particle_arrays):
+        iom = self.iom
+        io = iom.get_inlet_outlet(particle_arrays)
+        return io
+
+the function ``create_inlet_outlet`` takes the updater ``io`` created by the
+IOM and plugs it into the update routine of the application class.
+
+Overall setup
+~~~~~~~~~~~~~
+
+In order to run the simulation, the IOM object must be passed to the scheme.
+In the scheme, the IOM object must be implemented in the manner as described in
+:ref:`iom`.
+
+A few points to note while dealing with inlet outlet boundary condition,
+
+1. Construction of the IOM happens after the scheme is created with a
+``void`` IOM.
+
+.. code:: python
+
+    def create_scheme(self):
+        h = nu = None
+        s = EDACScheme(
+            ['fluid'], ['solid'], dim=2, rho0=rho, c0=c0, h=h, pb=p0,
+            nu=nu, inlet_outlet_manager=None,
+            inviscid_solids=['wall']
+        )
+        return s
+
+
+2. The IOM must be configured in the ``configure_scheme`` function.
+
+3. In case you change the integrator of the function, make sure the updater
+``io`` is updating in the appropriate stage. For example, in case of a
+``PECIntegrator`` class of integrator, the particles integrated half step in
+stage 1 and finally advected in stage 2 then ``io`` updates the particle
+arrays after stage 2 is complete. In case one wants to do the update in stage
+1 (while using another integrator) the arguments must be passed to the updater appropriately.
```

Comment:
How does one set the properties of the inlet/outlet?  Are those done with equations?  Is there an example that users should read?

Reply:
Added that in the Writing IOM section.